### PR TITLE
Install pyne, openmc and dagmc in a virtual environment

### PR DESCRIPTION
**Pull request by ahnaf-tahmid-chowdhury**
This pull request adds a script to install DAGMC, OpenMC and PyNE in a virtual environment named `nuclear-boy`. The script is intended to simplify the installation process for these dependencies and improve reproducibility. The script has been tested on Ubuntu 22.04 (Python 3.10.6) and should work on other Linux systems that support `apt-get` as well.

### CODE REVIEW

File: install_nuclear-boy.sh [hunk 1]
```diff
@@ -0,0 +1,351 @@
...
+  echo "Installing OpenMC..."
+  echo "--------------------"
+  cd ${env_dir}
+  git clone https://github.com/openmc-dev/openmc.git openmc-repo
+  cd openmc-repo
+  git checkout develop
+  mkdir bld
+  cd bld
+  # cmake, build and install
+  cmake .. -DCMAKE_INSTALL_PREFIX=${env_dir}
```
Comment (shimwell) [inline install_nuclear-boy.sh]: I just skimmed through this PR and noticed this compile command for openmc doesn't appear to make use of DAGMC

Comment (ahnaf-tahmid-chowdhury) [reply install_nuclear-boy.sh]: Thanks, I have also noticed that and updated. The `DAGMC` was not set.

**Comment by gonuke** [comment]
Thanks for this contribution.  We're planning to move this kind of thing into the main repo so it can be regularly tested and kept up to date.

We'll also have to think about how to organize contributed solutions like this.

**Comment by ahnaf-tahmid-chowdhury** [comment]
Thank you @gonuke  for getting back to me and for considering my contribution. I completely understand the need to ensure that the code is regularly tested and maintained, and I am happy to hear that my script can be considered for inclusion in the main repository.

I agree that organizing contributed solutions can be a challenge, and I am open to any suggestions you may have. In the meantime, please let me know if there is anything I can do to help with the integration of my script into the main repository.

**Pull request closed without merging**